src/rgw/rgw_sal_motr.cc

```diff
@@ -2034,10 +2035,8 @@ int MotrObject::MotrDeleteOp::delete_obj(const DoutPrefixProvider* dpp, optional
     }
     if (result.version_id == "")
       result.version_id = "null"; // show it as "null" in the reply
-  } else {
+  } else { // if (info.versioned())
     // Unversioned flow
```

Comment:
Hi, contradiction => line 2039 says `Unversioned flow` and like 2038 says `if (info.versioned())`.
should it be ` !info.versioned()` ??

Reply:
`// if (info.versioned())` was just a comment to let know that this is `else` block is related to which `if` block since there are lots of if-else blocks. Looks like this is creating more confusion so removing the comment.